Configure `azurerm_security_center_storage_defender` resource

patst opened the pull request:
<!--
Thank you for helping to improve Official Azure Provider!

Please read through https://git.io/fj2m9 if this is your first time opening a
Official Azure Provider pull request. Find us in https://crossplane.slack.com 
if you need any help contributing.
-->

### Description of your changes

This PR includes configuring the `azurerm_security_center_storage_defender` resource and adds the example manifest.

Tested it locally and works.


I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review. 
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

I built the provider locally and tested it with a local KIND cluster.

A successful uptest run: https://github.com/crossplane-contrib/provider-upjet-azure/actions/runs/9913683138

[contribution process]: https://git.io/fj2m9

turkenf commented:
Hey @patst,

Thanks a lot for this PR; we appreciate your contributions 🙌 

Could you also add the example manifest of the resource under the [examples](https://github.com/crossplane-contrib/provider-upjet-azure/tree/main/examples) folder? 

=> `examples/storage/v1beta1/securitycenterstoragedefender.yaml`

patst commented:
> Hey @patst,
> 
> Thanks a lot for this PR; we appreciate your contributions 🙌
> 
> Could you also add the example manifest of the resource under the [examples](https://github.com/crossplane-contrib/provider-upjet-azure/tree/main/examples) folder?
> 
> => `examples/storage/v1beta1/securitycenterstoragedefender.yaml`

@turkenf  thanks for the fast feedback.
I added the example.

I see the error related to the git diff after running the linters. I am not sure why it is there. 
If I run `make reviewable` locally it removes the redundant aliasses which seem to lead to the diff in the pipeline.

Do you have an idea why?

turkenf commented:
/test-examples="examples/storage/v1beta1/securitycenterstoragedefender.yaml"

turkenf reviewed (commented):
Hey @patst, I left a few comments for you to consider. Also please pay attention to my comment below.

Since the [azurerm_security_center_storage_defender](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/security_center_storage_defender) resource is a resource from the security group, it would be better to create it in the security group instead of the storage group. 
I think StorageDefender is more meaningful and sufficient, considering the kind of resource.
To do the above, you can add a configuration to the [config/security/config.go](https://github.com/crossplane-contrib/provider-upjet-azure/blob/00c9a189692e6952eaf06809a32756ef6a9e1810/config/security/config.go#L29) file as follows:
```
	p.AddResourceConfigurator("azurerm_security_center_storage_defender", func(r *config.Resource) {
		r.Kind = "StorageDefender"
		r.ShortGroup = "security"
	})
```

Please do not forget to move the example under the security group after making these changes.

turkenf commented:
Please also sign the commit: https://github.com/crossplane-contrib/provider-upjet-azure/pull/782/checks?check_run_id=27361327990

patst commented:
> Hey @patst, I left a few comments for you to consider. Also please pay attention to my comment below.
> 
> Since the [azurerm_security_center_storage_defender](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/security_center_storage_defender) resource is a resource from the security group, it would be better to create it in the security group instead of the storage group. I think StorageDefender is more meaningful and sufficient, considering the kind of resource. To do the above, you can add a configuration to the [config/security/config.go](https://github.com/crossplane-contrib/provider-upjet-azure/blob/00c9a189692e6952eaf06809a32756ef6a9e1810/config/security/config.go#L29) file as follows:
> 
> ```
> 	p.AddResourceConfigurator("azurerm_security_center_storage_defender", func(r *config.Resource) {
> 		r.Kind = "StorageDefender"
> 		r.ShortGroup = "security"
> 	})
> ```
> 
> Please do not forget to move the example under the security group after making these changes.

@turkenf thanks for the review!
I moved the resource to the security group and renamed it. I had to start from a fresh branch and force push, because the resource was not removed from the storage code after running `make generate`

Hope everything looks fine now, otherwise feel free to comment

turkenf commented:
/test-examples="examples/security/v1beta1/storagedefender.yaml"

turkenf commented:
/test-examples="examples/security/v1beta1/storagedefender.yaml"

turkenf reviewed (approved):
Thanks for the quick fixes and effort @patst 🙏 LGTM.
